comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -51,10 +62,14 @@ type resourceTags struct {
 func NewCollector() (workloadmeta.CollectorProvider, error) {
 	return workloadmeta.CollectorProvider{
 		Collector: &collector{
-			id:           collectorID,
-			resourceTags: make(map[string]resourceTags),
-			seen:         make(map[workloadmeta.EntityID]struct{}),
-			catalog:      workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+			id:                    collectorID,
+			resourceTags:          make(map[string]resourceTags),
+			seen:                  make(map[workloadmeta.EntityID]struct{}),
+			catalog:               workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+			taskCollectionEnabled: util.IsTaskCollectionEnabled(),
+			taskCache:             cache.New(config.Datadog.GetDuration("ecs_task_cache_ttl"), 30*time.Second),
+			taskRateRPS:           config.Datadog.GetInt("ecs_task_collection_rate"),
+			taskRateBurst:         config.Datadog.GetInt("ecs_task_collection_burst"),
```

Comment:
Could you declare the config component as a dependency for this component and fetch the values from the config component rather than from the `pkg/config`? 😄